src/lib.rs

```diff
@@ -308,6 +308,19 @@ impl Entry {
     /// Create an entry for the given service and user.
     ///
     /// The default credential builder is used.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an [`Error`] if the `service` or `user` values are invalid.
+    /// The specific reasons for invalidity are platform-dependent, but include length constraints.
+    ///
+    /// [`Error`]: Error
+    ///
+    /// # Panics
+    ///
+    /// This function may panic if the internal credential builder's `RwLock` is poisoned. This
+    /// would typically happen if a thread panicked while holding a lock. If you encounter this,
+    /// it is considered a serious bug and should be reported.
```

Comment:
I'd rather not get into why an `RwLock` might be poisoned.  How about something like this instead:

In the very unlikely event that the internal credential builder's `RwLock` is poisoned, this function
will panic.  If you encounter this, and especially if you can reproduce it, please report a bug with the
details (and preferably a backtrace) so the developers can investigate.